File: jwt-practice/src/main/java/com/example/jwtpractice2/exception/GlobalExceptionHandler.java
package com.example.jwtpractice2.exception;

import com.example.jwtpractice2.error.ErrorCode;
import com.example.jwtpractice2.exception.response.ExceptionResponse;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.ExceptionHandler;
import org.springframework.web.bind.annotation.RestControllerAdvice;

@RestControllerAdvice
public class GlobalExceptionHandler {

    @ExceptionHandler(BusinessException.class)
    protected ResponseEntity<ExceptionResponse> handleException(BusinessException e) {
        final ErrorCode errorCode = e.getErrorCode();
        return new ResponseEntity<>(new ExceptionResponse(errorCode.getMessage()), HttpStatus.valueOf(errorCode.getStatus()));
    }
}

File: issue-tracker/issue-service/src/main/kotlin/com/example/issuetracker/exception/GlobalExceptionHandler.kt
package com.example.issuetracker.exception

import com.auth0.jwt.exceptions.TokenExpiredException
import mu.KotlinLogging
import org.springframework.web.bind.annotation.ExceptionHandler
import org.springframework.web.bind.annotation.RestControllerAdvice

@RestControllerAdvice
class GlobalExceptionHandler {

    private val logger = KotlinLogging.logger {}

    @ExceptionHandler(ServerException::class)
    fun handleServerException(exception: ServerException): ErrorResponse {
        logger.error { exception.message }

        return ErrorResponse(
            code = exception.code,
            message = exception.message
        )
    }

    @ExceptionHandler(TokenExpiredException::class)
    fun handleTokenExpiredException(exception: TokenExpiredException): ErrorResponse {
        logger.error { exception.message }

        return ErrorResponse(
            code = 401,
            message = "Token Expired Error"
        )
    }

    @ExceptionHandler(Exception::class)
    fun handleException(exception: Exception): ErrorResponse {
        logger.error { exception.message }

        return ErrorResponse(
            code = 500,
            message = "Internal Server Error"
        )
    }
}